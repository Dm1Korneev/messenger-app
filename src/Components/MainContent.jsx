import React from 'react';
import PropTypes from 'prop-types';

// containers
import withStyles from '@material-ui/core/styles/withStyles';
import Grid from '@material-ui/core/Grid';
import MessagesList from 'Containers/MessagesList';
import MessageInput from 'Containers/MessageInput';

// @material-ui

function MainContent(props) {
  const { classes } = props;

  return (
    <main className={classes.main}>
      <Grid item xs={12} sm={11} md={10} lg={9} className={classes.grid}>
        <div className={classes.content}>
          <div className={classes.appBarSpacer} />
          <MessagesList />
          <MessageInput />
        </div>
      </Grid>
    </main>
  );
}

MainContent.propTypes = {
  classes: PropTypes.instanceOf(Object).isRequired,
};

const styles = (theme) => ({
  main: {
    flexGrow: 1,
  },
  grid: {
    margin: '0 auto',
    height: '100%',
  },
  appBarSpacer: { marginTop: theme.spacing.unit, ...theme.mixins.toolbar },
  content: {
    height: '100%',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
  },
});

export default withStyles(styles)(MainContent);
