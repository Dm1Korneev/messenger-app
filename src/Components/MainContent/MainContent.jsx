import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Box from '@material-ui/core/Box';
import Grid from '@material-ui/core/Grid';

import MessageInput from 'Components/MessageInput';
import MessagesList from 'Components/MessagesList';

const useStyles = makeStyles((theme) => ({
  grid: {
    margin: '0 auto',
    height: '100%',
  },
  appBarSpacer: { marginTop: theme.spacing(1), ...theme.mixins.toolbar },
}));

function MainContent() {
  const classes = useStyles();

  return (
    <Box>
      <Grid item xs={12} sm={11} md={10} lg={9} className={classes.grid}>
        <Box display="flex" flexDirection="column" justifyContent="space-between" height="100%">
          <Box className={classes.appBarSpacer} />
          <MessagesList />
          <MessageInput />
        </Box>
      </Grid>
    </Box>
  );
}

export default MainContent;
