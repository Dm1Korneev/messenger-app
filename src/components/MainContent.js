import React from "react";
import PropTypes from "prop-types";

// containers
import MessagesList from "../containers/MessagesList";
import MessageInput from "../containers/MessageInput";

// @material-ui
import withStyles from "@material-ui/core/styles/withStyles";
import Grid from "@material-ui/core/Grid";

function MainContent(props) {
  const { classes } = props;

  return (
    <main className={classes.main}>
      <Grid item xs={12} sm={11} md={10} lg={9} className={classes.grid}>
        <div className={classes.content}>
          <div className={classes.appBarSpacer} />
          <MessagesList />
          <MessageInput />
        </div>
      </Grid>
    </main>
  );
}

MainContent.propTypes = {
  classes: PropTypes.object.isRequired
};

const styles = theme => ({
  main: {
    flexGrow: 1
  },
  grid: {
    margin: "0 auto",
    height: "100%"
  },
  appBarSpacer: { marginTop: theme.spacing.unit, ...theme.mixins.toolbar },
  content: {
    height: "100%",
    display: "flex",
    flexDirection: "column",
    justifyContent: "space-between"
  }
});

export default withStyles(styles)(MainContent);
