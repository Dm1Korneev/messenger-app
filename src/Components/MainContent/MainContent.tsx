import Box from '@material-ui/core/Box';
import Grid from '@material-ui/core/Grid';
import { makeStyles } from '@material-ui/core/styles';

import { MessageInput } from 'Components/MessageInput';
import { MessagesList } from 'Components/MessagesList';

const useStyles = makeStyles((theme) => ({
  grid: {
    margin: '0 auto',
    height: '100%',
  },
  appBarSpacer: {
    marginTop: theme.spacing(1),
    ...theme.mixins.toolbar,
  },
}));

type MainContentProps = {
  activeChatId?: string
}

export const MainContent = ({ activeChatId }: MainContentProps) => {
  const classes = useStyles();

  return (
    <Box width="100%">
      <Grid item xs={12} sm={11} md={10} lg={9} className={classes.grid}>
        <Box display="flex" flexDirection="column" justifyContent="space-between" height="100%">
          <Box className={classes.appBarSpacer} />
          <MessagesList activeChatId={activeChatId} />
          {activeChatId && <MessageInput activeChatId={activeChatId} />}
        </Box>
      </Grid>
    </Box>
  );
};
