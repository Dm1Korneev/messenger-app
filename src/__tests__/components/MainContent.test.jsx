import React from 'react';

import MainContent from 'Components/MainContent';

jest.mock('Containers/MessagesList', () => global.mockComponent('MessagesList'));
jest.mock('Containers/MessageInput', () => global.mockComponent('MessageInput'));

jest.mock('@material-ui/core/Grid', () => global.mockComponent('Grid'));

const props = {};

describe('render MainContent component', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = global.mount(
      <MainContent {...props} />,
    );
  });

  test('component is render', () => {
    expect(wrapper.find('MainContent').length).toBe(1);
  });

  test('MessagesList subcomponent is render', () => {
    expect(wrapper.find({ originalcomponent: 'MessagesList' }).length).toBe(1);
  });

  test('MessageInput subcomponent is render', () => {
    expect(wrapper.find({ originalcomponent: 'MessageInput' }).length).toBe(1);
  });
});

describe('snapshot-test MainContent component', () => {
  test('Renders correct properties', () => {
    global.mountExpect(
      <MainContent {...props} />,
    ).toMatchSnapshot();
  });
});
