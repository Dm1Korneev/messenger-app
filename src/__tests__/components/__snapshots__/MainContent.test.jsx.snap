// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot-test MainContent component Renders correct properties 1`] = `
<WithStyles(MainContent)>
  <MainContent
    classes={
      Object {
        "appBarSpacer": "MainContent-appBarSpacer-3",
        "content": "MainContent-content-4",
        "grid": "MainContent-grid-2",
        "main": "MainContent-main-1",
      }
    }
  >
    <main
      className="MainContent-main-1"
    >
      <Component
        className="MainContent-grid-2"
        item={true}
        lg={9}
        md={10}
        sm={11}
        xs={12}
      >
        <div
          originalcomponent="Grid"
          testprops={
            Object {
              "children": <div
                className="MainContent-content-4"
              >
                <div
                  className="MainContent-appBarSpacer-3"
                />
                <Unknown />
                <Unknown />
              </div>,
              "className": "MainContent-grid-2",
              "item": true,
              "lg": 9,
              "md": 10,
              "sm": 11,
              "xs": 12,
            }
          }
        >
          <div
            className="MainContent-content-4"
          >
            <div
              className="MainContent-appBarSpacer-3"
            />
            <Component>
              <div
                originalcomponent="MessagesList"
                testprops={Object {}}
              />
            </Component>
            <Component>
              <div
                originalcomponent="MessageInput"
                testprops={Object {}}
              />
            </Component>
          </div>
        </div>
      </Component>
    </main>
  </MainContent>
</WithStyles(MainContent)>
`;
